### PR TITLE
fix: add optional chaining in AdvancedConfigOptions.tsx

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description
Bug Fix

Fixes https://github.com/devtron-labs/devtron/issues/4709         https://github.com/devtron-labs/devtron/issues/4712

## Type of change
- [x] Bug fix 

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/bulkEdits/bulkedits.service.tsx [hunk 1]
```diff
@@ -3,14 +3,16 @@ import { Routes } from '../../config'
 
 export function updateBulkList(request): Promise<any> {
     const { apiVersion } = request
-    const kind = request.kind.toLocaleLowerCase()
+    let kind = String(`${request.kind}`)
+    kind = kind?.toLocaleLowerCase()
```
Comment (eshankvaish) [inline src/components/bulkEdits/bulkedits.service.tsx:7]: - We don't need both String and `` operator
- what if the kind is undefined?

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/components/app/details/appDetails/AppDetails.tsx [hunk 2]
```diff
@@ -332,7 +332,9 @@ export const Details: React.FC<DetailsType> = ({
                         setIsInitialTimelineDataLoading(false)
                     }
                 },
-            )
+            ).catch((err) => {
+                showError(err);
```
Comment (AbhishekA1509) [inline src/components/app/details/appDetails/AppDetails.tsx:336]: use prettier

File: src/components/app/details/triggerView/TriggerView.tsx [hunk 3]
```diff
@@ -166,11 +166,15 @@ class TriggerView extends Component<TriggerViewProps, TriggerViewState> {
 
     // FIXME: Seems like its missing a error state
```
Comment (AbhishekA1509) [inline src/components/app/details/triggerView/TriggerView.tsx:167]: Can you now remove this comment?

File: src/components/bulkEdits/BulkEdits.tsx [hunk 4]
```diff
@@ -122,9 +122,13 @@ export default class BulkEdits extends Component<BulkEditsProps, BulkEditsState>
         let configJson: any = {}
         try {
             configJson = yamlJsParser.parse(this.state.codeEditorPayload)
+            if (configJson.kind === undefined) {
+                throw new Error('provide kind')
```
Comment (AbhishekA1509) [inline src/components/bulkEdits/BulkEdits.tsx:126]: We can make it a constant

File: src/components/bulkEdits/BulkEdits.tsx [hunk 5]
```diff
@@ -122,9 +122,13 @@ export default class BulkEdits extends Component<BulkEditsProps, BulkEditsState>
         let configJson: any = {}
         try {
             configJson = yamlJsParser.parse(this.state.codeEditorPayload)
+            if (configJson.kind === undefined) {
```
Comment (AbhishekA1509) [inline src/components/bulkEdits/BulkEdits.tsx:125]: Please verify if we can write as if (!configJson.kind) {}

File: src/components/bulkEdits/BulkEdits.tsx [hunk 6]
```diff
@@ -122,9 +122,13 @@ export default class BulkEdits extends Component<BulkEditsProps, BulkEditsState>
...
+                throw new Error('provide kind')
+            }
         } catch (error) {
-            // Invalid YAML, couldn't be parsed to JSON. Show error toast
-            toast.error('Invalid Yaml')
+            if (error.message === 'provide kind') {
+                toast.error('Invalid Yaml: Provide kind') // Invalid YAML, kind not defined. Show error toast
+            } else toast.error('Invalid Yaml') // Invalid YAML, couldn't be parsed to JSON. Show error toast
```
Comment (AbhishekA1509) [inline src/components/bulkEdits/BulkEdits.tsx:131]: please use prettier here

File: src/components/bulkEdits/BulkEdits.tsx [hunk 7]
```diff
@@ -164,9 +168,13 @@ export default class BulkEdits extends Component<BulkEditsProps, BulkEditsState>
         let configJson: any = {}
         try {
             configJson = yamlJsParser.parse(this.state.codeEditorPayload)
+            if (configJson.kind === undefined) {
```
Comment (AbhishekA1509) [inline src/components/bulkEdits/BulkEdits.tsx:171]: Can we remove this duplication by encapsulating it in a function?

File: src/components/bulkEdits/bulkedits.service.tsx [hunk 8]
```diff
@@ -3,14 +3,16 @@ import { Routes } from '../../config'
...
+    let kind = `${request.kind}`
+    kind = kind?.toLocaleLowerCase()
     const URL = `${apiVersion}/${kind} `
     return post(URL, request)
 }
 
 export function updateImpactedObjectsList(request): Promise<any> {
     const { apiVersion } = request
-    const kind = request.kind.toLocaleLowerCase()
+    let kind = `${request.kind}`
```
Comment (AbhishekA1509) [inline src/components/bulkEdits/bulkedits.service.tsx:14]: Don't use let chain the expressions and use const

File: src/components/v2/appDetails/AppDetails.component.tsx [hunk 9]
```diff
@@ -60,7 +60,11 @@ const AppDetailsComponent = ({
 
     useEffect(() => {
         if (appDetails?.appType === AppType.EXTERNAL_HELM_CHART && params.appId) {
-            getSaveTelemetry(params.appId)
+            try {
+                getSaveTelemetry(params.appId)
```
Comment (AbhishekA1509) [inline src/components/v2/appDetails/AppDetails.component.tsx:64]: Can you verify loaders?

File: src/components/bulkEdits/BulkEdits.tsx (hunk 6 above)
Comment (AbhishekA1509) [reply src/components/bulkEdits/BulkEdits.tsx]: Also verify viewType being set to FORM

**Pull request closed without merging**